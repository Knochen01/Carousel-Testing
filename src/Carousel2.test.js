import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Carousel from './Carousel'


it('should render on the screed',() => {
    render(<Carousel />)
})

test('if it Matches Snapshot', () => {
    const { asFragment} = render(<Carousel />)
    expect(asFragment).toMatchSnapshot()
})

it("works when you click on the right arrow",() => {
    const { queryByTestId, queryByAltText } = render(<Carousel />)

    // Expect the first image to show, but not the second.
    expect(queryByAltText('Photo by Richard Pasquarella on Unsplash')).toBeInTheDocument();
    expect(queryByAltText('Photo by Pratik Patel on Unsplash')).not.toBeInTheDocument()

    // move forward in the carousel

    const rightArrow = queryByTestId("right-arrow");
    fireEvent.click(rightArrow)

    // expect the second image to show but not the first
    expect(queryByAltText('Photo by Pratik Patel on Unsplash')).toBeInTheDocument()
    expect(queryByAltText('Photo by Richard Pasquarella on Unsplash')).not.toBeInTheDocument();
})

it("hides and shows arrows appropriately",() => {
    const { getByTestId} = render(<Carousel />)
    const leftArrow = getByTestId('left-arrow')
    const rightArrow = getByTestId('right-arrow')

    // expect the left arrow to be missing, but the right arrow to be present
    expect(leftArrow).toHaveClass('hidden')
    expect(rightArrow).not.toHaveClass('hidden')

    // Move forward, expect both arrows to exist
    fireEvent.click(rightArrow)

    expect(leftArrow).not.toHaveClass('hidden')
    expect(rightArrow).not.toHaveClass('hidden')

    // Move foward again, expect the right arrow to be missing
    fireEvent.click(rightArrow)

    expect(rightArrow).toHaveClass('hidden')
    expect(leftArrow).not.toHaveClass('hidden')


})