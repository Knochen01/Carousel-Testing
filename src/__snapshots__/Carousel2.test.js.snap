// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if it Matches Snapshot 1`] = `[Function]`;
